pynvjitlink/tests/test_pynvjitlink_api.py

```diff
@@ -94,6 +94,11 @@ def test_add_cubin_with_fatbin_error(device_functions_fatbin):
         nvjitlinker.add_cubin(device_functions_fatbin, name)
 
 
+@pytest.mark.xfail(
+    reason="Fails in conda due to using CUDA 12.2u2. This should be fixed "
+    "when requiring CUDA 12.3 nvjitlink library. See: "
+    "https://github.com/rapidsai/pynvjitlink/pull/33#issuecomment-1885182905"
+)
```

Comment:
The inability to write this conditionally makes me wonder if pynvjitlink should embed the CUDA version used at build-time as a compile-time constant accessible via a Python API. That would be very helpful for us for debugging in the future. Since we're statically linking our only recourse for debugging purposes right now is to check when a particular pynvjitlink package was built and then see what version of CUDA the build scripts used at that time, right?

I'm not proposing a fix in this PR since that's out of scope, but IIUC that would be a nice follow-up.

Reply:
This would be fine with me but I don't think it's in scope here. It'll also be ameliorated somewhat by the ["long term plan"](https://github.com/rapidsai/pynvjitlink/pull/33#discussion_r1449272016) to build both pip and conda packages with a single nvjitlink version, which will establish a correspondence between pynvjitlink versions and nvjitlink versions.

Reply:
I agree that it's not in scope here. The long term plan doesn't really help much though, except that in the long term we will hopefully always support the latest CUDA version so we can always ask users to update to the latest pynvjitlink version and hope that fixes their problems. But for debugging purposes in the context of a broader error it would help us narrow down issues in bug reports.

Reply:
Actually am hoping we not only provide the ability to do this kind of check, but we actually do the check as part of `import pynvjitlink` (and error if it is not compatible)

It's possible that won't be enough and we will need to add some runtime checks for when `pynvjitlink` is used (and even defer some cases to the user to check)

Reply:
Opened https://github.com/rapidsai/pynvjitlink/issues/38 to continue this discussion separately.